### DISCUSSION TITLE
Using xarray dataset as a reader

### DISCUSSION THREAD

**Discussion by amandineschaeffer** [OpenDrift/opendrift#1272] [General]
Hi,

I am trying to use the reader with reader_netCDF_CF_generic, and an xarray dataset, but it is not working.
I even tried to loop back:

reader_wind = reader_netCDF_CF_generic.Reader('aaa.nc') # Which works fine
ds = reader_wind.Dataset

reader_wind = reader_netCDF_CF_generic.Reader(ds) # Which does not work

Am I missing something?
Regards,
Amandine

**Comment by knutfrode** [comment]
Hi,

The possibility to provide Xarray datasets to the readers was introduced recently, about a month ago.
Does it help to update your OpenDrift version?

If not, what is the error message you get?

**Reply by amandineschaeffer** [reply]
Thanks so much for the fast answer. You are right, I am not using the latest version of Opendrift. I will let you know if I still have a problem when I switch. Regards
